tests/kafkatest/tests/core/kraft_upgrade_test.py

```diff
@@ -108,6 +108,27 @@ def run_upgrade(self, from_kafka_version):
         assert len(cluster_id) == 22
         assert self.kafka.check_protocol_errors(self)
 
+        # Ensure we can create another topic and produce/consume to/from it
+        new_topic_cfg = {
+            "topic": "test-topic-2",
+            "partitions": self.partitions,
+            "replication-factor": self.replication_factor,
+            "configs": {"min.insync.replicas": 2}
```

Comment:
Don't we want to consider `self.replication_factor` here? Maybe `min(2, self.replication_factor)`.

Reply:
Addressed in https://github.com/apache/kafka/pull/15451/commits/6776bdf9751198294de2f8c4cf7c97351f34f97a